openedx/core/djangoapps/user_authn/views/utils.py

```diff
@@ -108,3 +115,55 @@ def get_mfe_context(request, redirect_to, tpa_hint=None):
         'countryCode': country_code,
     })
     return context
+
+
+def _get_username_prefix(data):
+    """
+    Get the username prefix (name initials) based on the provided data.
+
+    Args:
+    - data (dict):  Registration payload.
+
+    Returns:
+    - str: Name initials or None.
+    """
+    username_regex_partial = settings.USERNAME_REGEX_PARTIAL
+
+    if 'first_name' in data and 'last_name' in data:
+        first_name_data = unidecode(data.get('first_name', '')).strip()
+        last_name_data = unidecode(data.get('last_name', '')).strip()
+        first_name = ''.join(re.findall(username_regex_partial, first_name_data))
+        last_name = ''.join(re.findall(username_regex_partial, last_name_data))
+        return (first_name[0].upper() + last_name[0].upper()) if first_name and last_name else None
+    elif 'name' in data and data.get('name', '').strip():
+        name = unidecode(data['name']).strip()
+        name_parts = name.split()
+        initials = [re.findall(username_regex_partial, word)[0] for word in name_parts[:2]]
+        name_initials = ''.join([word[0] for word in initials])
+        return name_initials.upper() if name_initials else None
+    return None
```

Comment:
I think there is a lot of code repetition and some extra conditions here that can be simplified. (I am sure there are other ways to simplify the code. Test this on different scenarios if you end up using this)

```
    full_name = ''
    if data.get('first_name', '').strip() and data.get('last_name', '').strip():
        full_name = f"{unidecode(data.get('first_name', ''))} {unidecode(data.get('last_name', ''))}"
    elif data.get('name', '').strip():
        full_name = unidecode(data['name'])

    if full_name.strip():
        full_name = re.findall(username_regex_partial, full_name)[0]
        name_initials = "".join([name_part[0] for name_part in full_name.split()[:2]])
        return name_initials.upper() if name_initials else None

    return None
```